lldb/tools/lldb-dap/src-ts/uri-launch-handler.ts

```diff
@@ -0,0 +1,37 @@
+import * as vscode from "vscode";
+
+export class LaunchUriHandler implements vscode.UriHandler {
+    async handleUri(uri: vscode.Uri) {
+        try {
+            const params = new URLSearchParams(uri.query);
+            if (uri.path == '/launch/config') {
+                const configJson = params.get("config");
+                if (configJson === null) {
+                    throw new Error("Missing `config` URI parameter");
+                }
+                // Build the debug config.
+                let debugConfig: vscode.DebugConfiguration = {
+                    type: 'lldb-dap',
+                    request: 'launch',
+                    name: '',
+                };
+                Object.assign(debugConfig, JSON.parse(configJson));
+                debugConfig.name = debugConfig.name || debugConfig.program || "Adhoc Launch";
```

Comment:
if the request is attach, then make `URL-based Attach` the name. 
Also, `URL-based Launch` might be a better name.